### DISCUSSION TITLE
local secope as default

### DISCUSSION THREAD

**Discussion by amsitlab** [teal-language/tl#529] [General]
why not use local scope as default, and using some keyword to defining global vars/fuction, like:
```
-- global 1
pub Point = record do
   ....
end

-- or
-- global 2
export Point = record do
 ...
end

-- local 
Point = record do

end
```

that will produce clean code then writing local keyword many time.

**Comment by lenscas** [comment]
Not sure if this is the official (or the complete) reason but using `local` stays inline with Lua, which for a language that wants to stay as close to lua as possible is a plus.

Also, personally not a fan of `pub`/ `export` in your examples. You generally spoken export something by returning it rather than making it global.

**Comment by hishamhm** [comment]
We won't go with local-by-default, as it can produce very subtle bugs when people make mistakes reusing the same name in different nested scopes.

Teal already disallows global-by-default, and we're about to close the final gap in that discussion with #543. So, basically, both locals and globals need to be declared explicitly.
